### PR TITLE
cmake: don't use cmake libraries to create link cli

### PR DESCRIPTION
**Pull request by NickeZ**

### CODE REVIEW

**Comment by benma** [comment]
Well done!

A detailed commit message will be highly appreciated :grin:

**Comment by NickeZ** [comment]
I've changed around some things and improved the commit message.

File: test/device-test/CMakeLists.txt [hunk 1]
```diff
@@ -125,16 +129,32 @@ foreach(name ${BB02_TEST_LIST})
...
+      secp256k1
+      bitbox02_rust_c
+      noiseprotocol
+      cryptoauthlib
+      base32
+      bignum # TODO: only eth
+      sha3   # TODO: Only eth
+      ctaes
+      c
+      asf4-drivers-min
```
Comment (conte91) [inline test/device-test/CMakeLists.txt]: These lines seem to be repeated pretty much everywhere - can we store them in a variable? I have the impression we're going to screw up this sequence next time we change the `CMakeLists.txt` and we'll be back to the starting point.

Comment (NickeZ) [reply test/device-test/CMakeLists.txt]: yeah that makes sense, as long as it works with the linking order and so on. The rust-library can't be in the variable for example.

Comment (NickeZ) [reply test/device-test/CMakeLists.txt]: did a variable.

**Pull request merged**